File: demo/demo/src/main/java/com/example/demo/models/ElevatorStatusModel.java
package com.example.demo.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class ElevatorStatusModel {
    int currentFloor;
    int destinationFloor;
    int id;
    ElevatorState elevatorState;

    public ElevatorStatusModel(int currentFloor, int destinationFloor, int id, ElevatorState elevatorState) {
        this.currentFloor = currentFloor;
        this.destinationFloor = destinationFloor;
        this.id = id;
        this.elevatorState = elevatorState;
    }
}

File: elevator_system/lib/models/ElevatorStatusModel.dart
class ElevatorStatusModel {
  final int currentFloor;
  final int destinationFloor;
  final int id;
  final String elevatorState;

  ElevatorStatusModel({
    required this.currentFloor,
    required this.destinationFloor,
    required this.id,
    required this.elevatorState,
  });

  factory ElevatorStatusModel.fromJson(Map<String, dynamic> json) {
    return ElevatorStatusModel(
      currentFloor: json['currentFloor'],
      destinationFloor: json['destinationFloor'],
      id: json['id'],
      elevatorState: json['elevatorState'],
    );
  }
}
